dockstore-integration-testing/src/test/java/io/dockstore/webservice/metrics/MetricsDataS3ClientIT.java

```diff
@@ -195,31 +257,105 @@ void testS3Prototype() throws IOException {
      * @throws IOException
      */
     @Test
-    void testGetMetricsDataPagination() throws IOException {
-        // No endpoint exists yet, but toolId, versionName, and platform should be provided as query parameters.
-        // Owner would be the authenticated user that invokes the endpoint
-        final String toolId = "#workflow/github.com/ENCODE-DCC/pipeline-container/encode-mapping-cwl";
-        final String versionName = "1.0";
-        final String platform = "terra";
-        final long ownerUserId = 1;
-        // metricsRequestBody is an example of what the JSON request body would look like. It contains metrics for a workflow execution
-        final File metricsRequestBodyFile = new File(ResourceHelpers.resourceFilePath("prototype-metrics-request-body.json"));
-        final String metricsRequestBody = Files.asCharSource(metricsRequestBodyFile, StandardCharsets.UTF_8).read();
-
-        // Create an object in S3 for the workflow
-        MetricsDataS3Client client = new MetricsDataS3Client(BUCKET_NAME, s3Client);
-
-        // Add 1001 objects for a workflow version and verify that we retrieve all 1001 objects
+    void testGetMetricsDataPagination() throws FileNotFoundException {
+        // Admin user
+        final ApiClient webClient = getOpenAPIWebClient(USER_2_USERNAME, testingPostgres);
+        final WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        final ExtendedGa4GhApi extendedGa4GhApi = new ExtendedGa4GhApi(webClient);
+        final String platform = Partner.TERRA.name();
+        final String description = "A single execution";
+        // This uses the request body from a file, which is a visual JSON example of what the request body looks like
+        BufferedReader metricsRequestBodyBufferedReader = new BufferedReader(new FileReader(ResourceHelpers.resourceFilePath("prototype-metrics-request-body.json")));
+        List<Execution> executions = Arrays.stream(GSON.fromJson(metricsRequestBodyBufferedReader, Execution[].class)).toList();
+
+        // Register and publish a workflow
+        final String workflowId = "#workflow/github.com/DockstoreTestUser2/dockstore_workflow_cnv";
+        final String workflowVersionId = "master";
+        Workflow workflow = workflowApi.manualRegister("github", "DockstoreTestUser2/dockstore_workflow_cnv", "/workflow/cnv.cwl", "", "cwl",
+                "/test.json");
+        workflow = workflowApi.refresh1(workflow.getId(), false);
+        workflowApi.publish1(workflow.getId(), CommonTestUtilities.createOpenAPIPublishRequest(true));
+
+        // Create 1001 S3 objects by calling the endpoint that submits metrics 1001 times for a workflow version and verify that we retrieve all 1001 objects
```

Comment:
good call on testing this, document that the 1000 objects is an AWS restriction (default?)
https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjects.html
Unfortunately, I don't see a constant for it https://sdk.amazonaws.com/java/api/2.0.0/constant-values.html